qemu/tests/netperf_udp_perf.py

```diff
@@ -0,0 +1,229 @@
+            vm.copy_files_to(src, destpath, nic_index=1)
+            logexec = params.get("log_guestinfo_exec", "bash")
+            output = server_ctl.cmd_output("%s %s" % (logexec, destpath))
+            logfile = open(path, "a+")
+            logfile.write(output)
+            logfile.close()
+    except process.CmdError as err:
+        test.cancel("test faild")
+
+
+@error_context.context_aware
+def start_test(server, server_ctl, host, client, resultsdir,
+               test_duration="20",
+               burst_time="1",
+               numbers_per_burst="1000 1500 2000 2500 3000",
+               ver_cmd=None,
+               params=None, server_cyg=None, test=None):
+    """
+    Start to test with different combination of burst_time and numbers_per_burst
+
+    """
+    if params is None:
+        params = {}
+
+    guest_ver_cmd = params.get("guest_ver_cmd", "uname -r")
+    fd = open("%s/netperf-udp-perf.result.%s.RHS" % (
+              resultsdir, time.time()), "w")
+
+    test.write_test_keyval({'kvm-userspace-ver': netperf_base.ssh_cmd(
+                            host, ver_cmd).strip()})
+    test.write_test_keyval({'guest-kernel-ver': netperf_base.ssh_cmd(
+                            server_ctl, guest_ver_cmd).strip()})
+    test.write_test_keyval({'session-length': test_duration})
+
+    fd.write('### kvm-userspace-ver : %s\n' % netperf_base.ssh_cmd(
+                                            host, ver_cmd).strip())
+    fd.write('### guest-kernel-ver : %s\n' % netperf_base.ssh_cmd(
+                                server_ctl, guest_ver_cmd).strip())
+    fd.write('### kvm_version : %s\n' % os.uname()[2])
+    fd.write('### session-length : %s\n' % test_duration)
+
+    error_context.context("Start Netserver on guest", logging.info)
+    netperf_version = params.get("netperf_version", "2.6.0")
+    client_path = "/tmp/netperf-%s/src/netperf" % netperf_version
+    server_path = "/tmp/netperf-%s/src/netserver" % netperf_version
+    logging.info("Netserver start cmd is '%s'" % server_path)
+    netperf_base.ssh_cmd(server_ctl, "pidof netserver || %s" % server_path)
+
+    base = params.get("format_base", "18")
+    fbase = params.get("format_fbase", "2")
+    pid = str(os.getpid())
+    fname = "/tmp/netperf.%s.nf" % pid
+    numa_enable = params.get("netperf_with_numa", "yes") == "yes"
+
+    def netperf_thread(numa_enable, burst_time, numbers_per_burst,
+                       client, server, test_duration):
+        cmd = ""
+        if numa_enable:
+            n = abs(int(params.get("numa_node"))) - 1
+            cmd += "numactl --cpunodebind=%s --membind=%s " % (n, n)
+        cmd += "%s -t UDP_STREAM -w %s -b %s -H %s -l %s" % (
+                client_path, burst_time, numbers_per_burst, server, test_duration)
+        cmd += " >> %s" % fname
+        logging.info("Start netperf thread by cmd '%s'" % cmd)
+        netperf_base.ssh_cmd(client, cmd)
```

Comment:
This one is also very same with netperf.py, only difference is the command parameter, which could be a parameter of the function. like: 
netper_base.py
```
def netperf_thread(numa_enable, client, option)
        cmd = ""
￼        if numa_enable:
￼            n = abs(int(params.get("numa_node"))) - 1
￼            cmd += "numactl --cpunodebind=%s --membind=%s " % (n, n)
￼        cmd += option
￼        cmd += " >> %s" % fname
￼        logging.info("Start netperf thread by cmd '%s'" % cmd)
￼        ssh_cmd(client, cmd)
```
netperf_udp_perf.py
```

option = "%s -t UDP_STREAM -w %s -b %s -H %s -l %s" % (
￼                client_path, burst_time, numbers_per_burst, server, test_duration)
netperf_base.netperf_thread(numa_enable, client, option)
```

Reply:
updated.